setup.py

```diff
@@ -89,6 +90,13 @@ def find_version(*file_paths):
     'main_loop-log_memory': [
         'pympler',
         'matplotlib'
+    ],
+    # these are other cylc components within the cylc meta-package
+    'uiserver': [
+        'cylc-uiserver==0.3.0'
+    ],
```

Comment:
Doesn't it mean `cylc` has a downstream dependency on `cylc-uiserver`, and that `cylc-uiserver` has an upstream dependency on `cylc` now? Creating a circular dependency here?

Not sure whether it will always work (i.e. `pip`, `poetry`, `pipenv`, `tox`, `setuptools`, `importlib`; each of these could fail to handle a circular dependency I think? Looks like `pip` has had some issues before - https://github.com/pypa/pip/issues?q=circular+dependency)

Reply:
Good point, I was thinking (not very deeply!) that we'd have to remove the cylc dependency from the UIS, but that doesn't make sense since it is a real code dependency.

Reply:
I think we are re-using the protobuf schema and some GraphQL code too (also some ID parsing maybe?).

Reply:
The UI Server imports the Protobuf and GraphQL schema from cylc-flow directly.

Reply:
Not sure whether circular imports are problematic, will have to do some reading.